crates/oxc_linter/src/rules/jsx_a11y/role_has_required_aria_props.rs

```diff
@@ -0,0 +1,154 @@
+use crate::{context::LintContext, rule::Rule, utils::has_jsx_prop_lowercase, AstNode};
+use once_cell::sync::Lazy;
+use oxc_ast::{
+    ast::{JSXAttributeItem, JSXAttributeValue},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use phf::phf_map;
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+    "eslint-plugin-jsx-a11y(role-has-required-aria-props): `{role}` role is missing required aria props `{props}`."
+)]
+#[diagnostic(
+    severity(warning),
+    help("Add missing aria props `{props}` to the element with `{role}` role.")
+)]
+struct RoleHasRequiredAriaPropsDiagnostic {
+    #[label]
+    pub span: Span,
+    pub role: String,
+    pub props: String,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct RoleHasRequiredAriaProps;
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforces that elements with ARIA roles must have all required attributes for that role.
+    ///
+    /// ### Why is this bad?
+    /// Certain ARIA roles require specific attributes to express necessary semantics for assistive technology.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// <div role="checkbox" />
+    ///
+    /// // Good
+    /// <div role="checkbox" aria-checked="false" />
+    /// ````
+    RoleHasRequiredAriaProps,
+    correctness
+);
+
+static ROLE_TO_REQUIRED_ARIA_PROPS: Lazy<phf::Map<&'static str, &'static str>> = Lazy::new(|| {
+    phf_map! {
+        "checkbox" => "aria-checked",
+        "radio" => "aria-checked",
+        "combobox" => "aria-controls, aria-expanded",
+        "tab" => "aria-selected",
+        "slider" => "aria-valuemax, aria-valuemin, aria-valuenow",
+        "scrollbar" => "aria-valuemax, aria-valuemin, aria-valuenow, aria-orientation, aria-controls",
+        "heading" => "aria-level",
+        "option" => "aria-selected",
+    }
+});
+
+impl Rule for RoleHasRequiredAriaProps {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::JSXOpeningElement(jsx_el) = node.kind() {
+            let role_prop = match has_jsx_prop_lowercase(jsx_el, "role") {
+                Some(role_prop) => role_prop,
+                None => return,
+            };
+            let attr = match role_prop {
+                JSXAttributeItem::Attribute(attr) => attr,
+                JSXAttributeItem::SpreadAttribute(_) => return,
+            };
+            let role_values = match &attr.value {
+                Some(JSXAttributeValue::StringLiteral(role_values)) => role_values,
+                _ => return,
+            };
```

Comment:
these matches could become `let...else`
```rs
let Some(role_prop) = has_jsx_prop_lowercase(jsx_el, "role") else {return}
```

Reply:
Thanks for the review!
I can write more concisely there! Fixed it at https://github.com/oxc-project/oxc/pull/1881/commits/f4b41532736dbdc5cbe734250774d71cdeaf3525! 🙏